### PR TITLE
Updated 'checkout-deps' to check for 'pip'/'pip3' prior to installing 'AMBuild'

### PR DESCRIPTION
**Pull request by WildCard65**
## checkout-deps.sh:
   - Checks for an active 'python'/'python3' command.
   - Checks for an active pip installation in the detected Python installation, otherwise, it installs it via 'get-pip.py'

See: alliedmodders/ambuild#68

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ matrix:
         - MATRIX_EVAL="CC=clang && CXX=clang++"
 
     - os: osx
-      osx_image: xcode6.4
```
Comment (dvander) [inline .travis.yml:35]: This must not be changed, we test on the lowest version available (we support macOS 10.9 w/ XCode 6.2)

Comment (WildCard65) [reply .travis.yml:35]: Two things about that change:
1) Travis has deprecated it.
2) It has an invalid SSL configuration that prevents it from downloading 'get-pip.py'

Comment (WildCard65) [reply .travis.yml:35]:
```
Running builds with Xcode 6.4 in Travis CI is deprecated and will be removed in January 2019.
If Xcode 6.4 is critical to your builds, please contact our support team at support@travis-ci.com to discuss options.
```

```
Connecting to bootstrap.pypa.io|151.101.192.175|:443... connected.
ERROR: cannot verify bootstrap.pypa.io's certificate, issued by 'CN=GlobalSign CloudSSL CA - SHA256 - G3,O=GlobalSign nv-sa,C=BE':
  Issued certificate has expired.
To connect to bootstrap.pypa.io insecurely, use `--no-check-certificate'.
```

**Comment by WildCard65** [comment]
@dvander xcode 7.3 is the lowest I can go that's supported by Travis-CI (and won't bogusly error the build)

**Comment by dvander** [comment]
What do you mean bogusly error? It was working before no?

**Comment by WildCard65** [comment]
It was, but xcode-6.4 doesn't come with pip pre-installed, this forces this new script to download 'get-pip.py'.
The issue now is that xcode 6.4 MacOS mistakenly believes the SSL certificate is expired and fails the build.

**Comment by dvander** [comment]
Is it the download of get-pip.py that fails, or stuff inside get-pip.py? If the former, we can just include it in the repo. If the latter, can we fallback to setup.py? We put a fair bit of work into getting ancient compiler coverage (since that's what we have to build on, ultimately), and it'd be unfortunate to lose that.

**Comment by WildCard65** [comment]
Former, attached is a transcript of the headache Travis caused for me. Another thing, you'll need to contact Travis themselves (if you haven't done so in the past year and a half?) for xcode-6.4 related options.

```
Connecting to bootstrap.pypa.io|151.101.192.175|:443... connected.
ERROR: cannot verify bootstrap.pypa.io's certificate, issued by 'CN=GlobalSign CloudSSL CA - SHA256 - G3,O=GlobalSign nv-sa,C=BE':
  Issued certificate has expired.
To connect to bootstrap.pypa.io insecurely, use `--no-check-certificate'.
```

**Comment by dvander** [comment]
Use --no-check-certificate then? (Or bundle it in the repo, as if it were a prebuilt.)

**Comment by WildCard65** [comment]
I'll give the argument a try (I attempted to, but I may not of actually committed the change when I did it)

**Comment by WildCard65** [comment]
@dvander We will still get the warning that xcode-6.4 is deprecated and slated for removal (January last year??)

**Comment by WildCard65** [comment]
And ran into another issue, the paths for the user install of 'ambuild' is different from Linux

**Comment by dvander** [comment]
Sounds like they're behind schedule. I guess, upgrading to 7.3 is fine. Note that our official builds are on _xcode 6.2_. checkout-deps.sh doesn't have to work there, but ambuild does.

**Comment by WildCard65** [comment]
Well, the warning Travis is spewing DOES mention that you can contact them for options if xcode6.4 is REQUIRED.

**Comment by WildCard65** [comment]
Build completed with success... yay...

Now how to deal with the deprecation warning, you can either:
  A) contact Travis for options (if you want to stick with xcode6.4 or down to xcode6.2)
  B) we can hop to xcode7.3 and remove the ignore SSL options.

**Comment by dvander** [comment]
Thanks! We can upgrade once they kill it off... there must be some paying customer keeping it alive for now.

I would very much love to bump our minimum macOS version to support C++17. Maybe the upcoming transition to ARM will make that possible.

**Pull request merged**